### ISSUE TITLE
`max_native_zoom` property for Providers 

### ISSUE THREAD

**Issue by vpipkt** [geopandas/contextily#161]
Allow for a `max_native_zoom` property in Provider object, which would have the same semantics as [`maxNativeZoom` in Leaflet](https://leafletjs.com/reference-1.2.0.html#gridlayer-maxnativezoom). Honoring this option would require auto-scaling the returned image at `max_native_zoom` to the requested zoom.

Implementation seems like it would  entail changes in tile.py especially `_validate_zoom`, `bounds2img`, and possibly ` _fetch_tile`. In `bounds2img` and `_validate_zoom, logic to track the requested zoom from the caller and the `z` to pass to the provider. And it seems reasonable to use `PIL.Image.resize`in `bounds2img` while the data are still in a PIL.Image. 

I am happy to propose a PR, but I'm curious to hear what others may have to think about this idea. I am pretty new to contextily so I'm curious if there might be unintended consequences.

**Comment by vpipkt** [comment]
Somewhat related: https://github.com/leaflet-extras/leaflet-providers/pull/369/files 

If that PR is merged I suppose I will make a PR against contextily/_providers.py

**Comment by jorisvandenbossche** [comment]
What is exactly the difference between `max_zoom` and `max_native_zoom` ?

**Comment by vpipkt** [comment]
In leaflet, [`maxZoom`](https://leafletjs.com/reference-1.2.0.html#gridlayer-maxzoom), is "the maximum zoom level up to which this layer will be displayed (inclusive)." At higher zoom levels the default behavior is to display an empty / transparent image for that layer. In some instances the map will just be a grey box.

In leaflet, [`maxNativeZoom`](https://leafletjs.com/reference-1.2.0.html#gridlayer-maxnativezoom) is "Maximum zoom number the tile source has available. If it is specified, the tiles on all zoom levels higher than maxNativeZoom will be loaded from maxNativeZoom level and auto-scaled." Instead of a transparent or grey display, it will upsample the result from the specified `maxNativeZoom`. 

Basically specifying maxNativeZoom allows you to set a maximum on the zoom level the _client_ requests; then the client knows to auto scale in cases where the state demands a higher zoom level.

**Comment by vpipkt** [comment]
PR closed... issue filed with stamen maps: https://github.com/stamen/maps.stamen.com/issues/162 

> Somewhat related: https://github.com/leaflet-extras/leaflet-providers/pull/369/files
> 
> If that PR is merged I suppose I will make a PR against contextily/_providers.py

**Comment by vpipkt** [comment]
oops made a comment that was meant for a PR in another repo. @jorisvandenbossche does this sound reasonable enough that a PR would be considered?
